back/dora/users/tests/test_views.py

```diff
@@ -0,0 +1,129 @@
+                "consent_version": "1.1",
+                "consent_choices": {
+                    "google_cse": True,
+                    "matomo": True,
+                },
+            },
+        )
+
+        self.assertEqual(response.status_code, 201)
+
+        consent_record = ConsentRecord.objects.last()
+
+        self.assertEqual(
+            consent_record.anonymous_user_hash, "057c49070eb7e2504610edbdab56ea00"
+        )
+        self.assertEqual(consent_record.user, None)
+        self.assertEqual(
+            consent_record.consent_version,
+            "1.1",
+        )
+        self.assertEqual(
+            consent_record.consent_choices,
+            {
+                "google_cse": True,
+                "matomo": True,
+            },
+        )
+
+    def test_create_consent_record_for_authenticated_user(self):
+        user = make_user()
+        self.client.force_authenticate(user=user)
+
+        response = self.client.post(
+            "/consent-record/",
+            {
+                "anonymous_user_hash": "057c49070eb7e2504610edbdab56ea00",
+                "consent_version": "1.2",
+                "consent_choices": {
+                    "google_cse": False,
+                    "matomo": False,
+                },
+            },
+        )
+
+        self.assertEqual(response.status_code, 201)
+
+        consent_record = ConsentRecord.objects.last()
+
+        self.assertEqual(consent_record.user, user)
+        self.assertEqual(
+            consent_record.anonymous_user_hash, "057c49070eb7e2504610edbdab56ea00"
+        )
+        self.assertEqual(
+            consent_record.consent_version,
+            "1.2",
+        )
+        self.assertEqual(
+            consent_record.consent_choices,
+            {
+                "google_cse": False,
+                "matomo": False,
+            },
+        )
+
+    def test_raise_validation_error_when_neither_user_nor_anonymous_user_hash(self):
+        response = self.client.post(
+            "/consent-record/",
+            {"consent_version": "1.1", "consent_choices": {}},
+        )
+
+        self.assertEqual(response.status_code, 400)
+
+    def test_raise_validation_error_when_consent_choices_not_a_dict(self):
+        response = self.client.post(
+            "/consent-record/",
+            {
+                "anonymous_user_hash": "12345abcdef",
+                "consent_version": "1.2",
+                "consent_choices": "abc",
+            },
+        )
+
+        self.assertEqual(response.status_code, 400)
+
+    def test_raise_validation_error_when_consent_choices_invalid(self):
+        response = self.client.post(
+            "/consent-record/",
+            {
+                "anonymous_user_hash": "12345abcdef",
+                "consent_version": "1.2",
+                "consent_choices": {
+                    "google_cse": "abc",
+                    "matomo": False,
+                },
+            },
+        )
```

Comment:
Pour ces 2 tests faut utiliser la bonne longueur pour `anonymous_user_hash` sinon c'est pas sûr qu'on test ce qu'on pense :).
```suggestion
    def test_raise_validation_error_when_consent_choices_not_a_dict(self):
        response = self.client.post(
            "/consent-record/",
            {
                "anonymous_user_hash": "057c49070eb7e2504610edbdab56ea00",
                "consent_version": "1.2",
                "consent_choices": "abc",
            },
        )

        self.assertEqual(response.status_code, 400)

    def test_raise_validation_error_when_consent_choices_invalid(self):
        response = self.client.post(
            "/consent-record/",
            {
                "anonymous_user_hash": "057c49070eb7e2504610edbdab56ea00",
                "consent_version": "1.2",
                "consent_choices": {
                    "google_cse": "abc",
                    "matomo": False,
                },
            },
        )
```